### PR TITLE
bip21: support bitcoin uri scheme

### PR DESCRIPTION
**Pull request by lvaccaro**
update uri interface to support bip23 uri as: `/bitcoin/bitcoin:<address>?amount=<decimal>&label=<text>` .

- update webserver interface
- add bip21 struct + serialization/deserialization with url encoding
- add new section to html page if specified in the bip21 as the following image

Screenshot 2021-05-05 at 02 54 25

### CODE REVIEW

File: src/bip21.rs [hunk 1]
```diff
@@ -0,0 +1,102 @@
+use percent_encoding::{utf8_percent_encode, AsciiSet, CONTROLS};
+use std::collections::BTreeMap;
+use url::{ParseError, Url};
+
+pub struct Bip21 {
+    pub scheme: String,
+    pub address: String,
+    pub amount: Option<String>,
```
Comment (RCasatta) [inline src/bip21.rs:8]: did you consider using `bitcoin::Address` and `bitcoin::Amount` here?

Comment (lvaccaro) [reply src/bip21.rs]: +1

**Comment by notmandatory** [comment]
ACK 9113c10 . Nice improvement! I verified the QR Code displayed and the URI expected match. Also tested that going to URL:

```
http://0.0.0.0:8080/bitcoin?amount=10&label=test%20test
```

redirects to the expected next unused address URL:

```
http://0.0.0.0:8080/bitcoin/bitcoin:tb1q32a23q6u3yy89l8svrt80a54h06qvn7gnuvsen?amount=10&label=test%20test
```

**Comment by lvaccaro** [comment]
refactored in #36

**Pull request closed without merging**